plenum/bls/bls_bft_replica_plenum.py

```diff
@@ -198,6 +266,29 @@ def _sign_state(self, pre_prepare: PrePrepare):
     def _can_calculate_multi_sig(self,
                                  key_3PC,
                                  quorums) -> bool:
+        if key_3PC in self._all_signatures:
+            sigs_for_request = self._all_signatures[key_3PC]
+            sigs_invalid = list(
+                filter(
+                    lambda item: not quorums.bls_signatures.is_reached(len(list(item[1].values()))),
+                    sigs_for_request.items()
+                )
+            )
+            if sigs_invalid:
+                self._all_signatures.pop(key_3PC, None)
```

Comment:
Method name `can_do_something` needs to do only the checks, not modifications. So, code like `self._all_signatures.pop` looks really suspicious here.